src/module-elasticsuite-catalog/Model/ProductSorter/AbstractPreview.php

```diff
@@ -147,7 +155,10 @@ private function getProductCollection()
      */
     private function getUnsortedProductData()
     {
-        $productCollection = $this->getProductCollection()->setPageSize($this->size);
+        $productCollection = $this->getProductCollection()
+            ->setSearchQuery($this->search)
```

Comment:
Shouldn't we apply `setSearchQuery` only if `$this->search !== ''`` ?

Reply:
Possibly, but further down the line, the fulltext query part is not generated if `$this->search !== ''``

Reply:
The receiving method should check its input and act on that. Now you have fluent code. Otherwise you'd end up with checks on both side which isn't necessary.